File: webforj-components/webforj-icons/src/main/java/com/webforj/component/icons/DwcIcon.java
package com.webforj.component.icons;

import java.util.Locale;

/**
 * Enumeration of all Dwc core icons.
 *
 * <p>
 * DWC icons are a small subset of the Tabler icons.
 * </p>
 *
 * @see <a href="https://tablericons.com/">Tabler Icons</a>
 *
 * @author Hyyan Abo Fakher
 * @since 24.11
 */
public enum DwcIcon implements IconFactory {
  CALENDAR, CHECK, CHEVRON_DOWN, CHEVRON_LEFT, CHEVRON_RIGHT, CHEVRON_UP, CHEVRONS_LEFT, CHEVRONS_RIGHT, EYE_OFF, EYE, ICON_X, PLUS, MINUS, SQUARE_PLUS, SQUARE_MINUS, FILE, FOLDER, FOLDER_OPENED, STOP, QUESTION, WARNING, INFO, COLOR_PICKER, COPY, CHECKS, UPLOAD, REFRESH, PLAY, ANIMATED_SPINNER, ARROW_DOWN, ARROW_LEFT, ARROW_RIGHT, ARROW_UP, TYPOGRAPHY, GRIP_HORIZONTAL, GRIP_VERTICAL, LIST, GRID, HARD_DRIVE, COLUMNS, TIME;

  /**
   * {@inheritDoc}
   */
  @Override
  public Icon create() {
    return new Icon(String.valueOf(this), getPool());
  }

  /**
   * {@inheritDoc}
   */
  @Override
  public String toString() {
    return name().toLowerCase(Locale.ENGLISH).replace('_', '-');
  }
}

File: webforj-kotlin/src/main/kotlin/com/webforj/kotlin/dsl/component/icons/DwcIcon.kt
package com.webforj.kotlin.dsl.component.icons

import com.webforj.component.icons.DwcIcon
import com.webforj.component.icons.Icon
import com.webforj.concern.HasComponents
import com.webforj.kotlin.dsl.WebforjDsl
import com.webforj.kotlin.dsl.init

/**
 * Creates a DWC `Icon`.
 * ```
 * ... {
 *  dwcIcon(DwcIcon.CHECK)
 *  dwcIcon(DwcIcon.PLUS) {
 *    theme = Theme.SUCCESS
 *  }
 * }
 * ```
 *
 * @param icon The `Icon` to create.
 * @param block The initialization steps of the `Icon`.
 * @return The configured `Icon`.
 * @see DwcIcon
 */
fun @WebforjDsl HasComponents.dwcIcon(icon: DwcIcon, block: @WebforjDsl Icon.() -> Unit = {}): Icon = init(icon.create(), block)
